whole-tale/gwvolman
# Recorded run task

**This PR**
* Implements the `recorded_run` task which:
   * Builds the image for the specified version/run
   * Mounts workspace and data for the specified version/run
   * Executes `reprozip --trace ... ./run.sh` using the user's image, which generates output in the run workspace `.wholetale` directory
   * Executes CPR reports which generate output into `.wholetale/.cpr` directory
* Also
  * Moves volume creation/mounting code into helper functions that can be used by new recorded_run task
  * Modifies _build_image helper to enable use by recorded_run task
  * Adds tests

**To Test**
* Depends on
  * https://github.com/whole-tale/wt_versioning/pull/29
  * https://github.com/whole-tale/repo2docker_wholetale/pull/21
  * https://github.com/whole-tale/wt-cpr
* Create a tale with an executable `run.sh` in the workspace root.  
   * TODO: This needs to change with the run configuration implementation
* Create a version and run (`POST /run`) for the version
* Change the main workspace in a way that would be detectable during the run (i.e., modify run.sh)
* Optionally `docker logs -f celery_worker` to see what's happening and any errors
* `POST /run/{id}/start`
* Confirm the following
  * Image is built using the run workspace for the selected version
  * `reprozip trace` executes successfully (`.wholetale/.reprozip-trace/trace.sqlite3` is present in run workspace
  *  `cpr` reports execute successfully (`.wholetale/.cpr/` contains `exec.txt`, `write.txt`, `read.txt`)

Note: The reports are POC for now.
